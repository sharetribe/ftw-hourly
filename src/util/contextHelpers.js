import React, { Component as ReactComponent } from 'react';
import PropTypes from 'prop-types';
import throttle from 'lodash/throttle';

/**
 * A higher order component (HOC) to take the togglePageClassNames function from
 * the context that the Page component has provided.
 */
export const withTogglePageClassNames = Component => {
  const WithTogglePageClassNamesComponent = (props, context) => (
    <Component togglePageClassNames={context.togglePageClassNames} {...props} />
  );

  WithTogglePageClassNamesComponent.displayName = `withTogglePageClassNames(${Component.displayName ||
    Component.name})`;

  const { func } = PropTypes;

  WithTogglePageClassNamesComponent.contextTypes = {
    togglePageClassNames: func.isRequired,
  };

  return WithTogglePageClassNamesComponent;
};

/**
 * A higher order component (HOC) that provides the current viewport
 * dimensions to the wrapped component as a `viewport` prop that has
 * the shape `{ width: 600, height: 400}`.
 */
export const withViewport = Component => {
  // The resize event is flooded when the browser is resized. We'll
  // use a small timeout to throttle changing the viewport since it
  // will trigger rerendering.
  const WAIT_MS = 100;

  class WithViewportComponent extends ReactComponent {
    constructor(props) {
      super(props);
      this.state = { width: 0, height: 0 };
      this.handleWindowResize = this.handleWindowResize.bind(this);
      this.setViewport = throttle(this.setViewport.bind(this), WAIT_MS);
    }
    componentDidMount() {
      this.setViewport();
      window.addEventListener('resize', this.handleWindowResize);
      window.addEventListener('orientationchange', this.handleWindowResize);
    }
    componentWillUnmount() {
      window.removeEventListener('resize', this.handleWindowResize);
      window.removeEventListener('orientationchange', this.handleWindowResize);
    }
    handleWindowResize() {
      this.setViewport();
    }
    setViewport() {
      this.setState({
        width: window.innerWidth,
        height: window.innerHeight,
      });
    }
    render() {
      const viewport = this.state;
      const props = { ...this.props, viewport };
      return <Component {...props} />;
    }
  }

  WithViewportComponent.displayName = `withViewport(${Component.displayName || Component.name})`;

  return WithViewportComponent;
};

/**
 * A higher order component (HOC) that provides dimensions to the wrapped component as a
 * `dimensions` prop that has the shape `{ width: 600, height: 400}`.
 *
 * @param {React.Component} Component to be wrapped by this HOC
 * @param {Object} options pass in options like maxWidth and maxHeight.
 *
 * @return {Object} HOC component which knows its dimensions
 */
export const withDimensions = (Component, options = {}) => {
  // The resize event is flooded when the browser is resized. We'll
  // use a small timeout to throttle changing the viewport since it
  // will trigger rerendering.
  const THROTTLE_WAIT_MS = 200;
  // First render default wait after mounting (small wait for styled paint)
  const RENDER_WAIT_MS = 100;

  class WithDimensionsComponent extends ReactComponent {
    constructor(props) {
      super(props);
      this.element = null;
      this.defaultRenderTimeout = null;

      this.state = { width: 0, height: 0 };

      this.handleWindowResize = throttle(this.handleWindowResize.bind(this), THROTTLE_WAIT_MS);
      this.setDimensions = this.setDimensions.bind(this);
    }

    componentDidMount() {
      window.addEventListener('resize', this.handleWindowResize);
      window.addEventListener('orientationchange', this.handleWindowResize);

      this.defaultRenderTimeout = window.setTimeout(() => {
        this.setDimensions();
      }, RENDER_WAIT_MS);
    }

    componentWillUnmount() {
      window.removeEventListener('resize', this.handleWindowResize);
      window.removeEventListener('orientationchange', this.handleWindowResize);
      window.clearTimeout(this.defaultRenderTimeout);
    }

    handleWindowResize() {
      window.requestAnimationFrame(() => {
        this.setDimensions();
      });
    }

    setDimensions() {
      this.setState(prevState => {
        const { clientWidth, clientHeight } = this.element || { clientWidth: 0, clientHeight: 0 };
        return { width: clientWidth, height: clientHeight };
      });
    }

    render() {
      // Dimensions from state (i.e. dimension after previous resize)
      // These are needed for component rerenders
      const { width, height } = this.state;

      // Current dimensions from element reference
      const { clientWidth, clientHeight } = this.element || { clientWidth: 0, clientHeight: 0 };
      const hasDimensions =
        (width !== 0 && height !== 0) || (clientWidth !== 0 && clientHeight !== 0);

      // clientWidth and clientHeight
      const currentDimensions =
        clientWidth !== 0 && clientHeight !== 0
          ? { width: clientWidth, height: clientHeight }
          : width !== 0 && height !== 0
            ? { width, height }
            : {};

      const props = { ...this.props, dimensions: currentDimensions };

      // lazyLoadWithDimensions HOC needs to take all given space
      // unless max dimensions are provided through options.
      const { maxWidth, maxHeight } = options;
      const maxWidthMaybe = maxWidth ? { maxWidth } : {};
      const maxHeightMaybe = maxHeight ? { maxHeight } : {};
      const style =
        maxWidth || maxHeight
          ? { width: '100%', height: '100%', ...maxWidthMaybe, ...maxHeightMaybe }
          : { position: 'absolute', top: 0, right: 0, bottom: 0, left: 0 };

      return (
        <div
          ref={element => {
            this.element = element;
          }}
          style={style}
        >
          {hasDimensions ? <Component {...props} /> : null}
        </div>
      );
    }
  }

  WithDimensionsComponent.displayName = `withDimensions(${Component.displayName ||
    Component.name})`;

  return WithDimensionsComponent;
};

/**
 * A higher order component (HOC) that lazy loads the current element and provides
 * dimensions to the wrapped component as a `dimensions` prop that has
 * the shape `{ width: 600, height: 400}`.
 *
 * @param {React.Component} Component to be wrapped by this HOC
 * @param {Object} options pass in options like maxWidth and maxHeight. To load component after
 * initial rendering has passed or after user has interacted with the window (e.g. scrolled),
 * use`loadAfterInitialRendering: 1500` (value should be milliseconds).
 *
 * @return {Object} HOC component which knows its dimensions
 */
export const lazyLoadWithDimensions = (Component, options = {}) => {
  // The resize event is flooded when the browser is resized. We'll
  // use a small timeout to throttle changing the viewport since it
  // will trigger rerendering.
  const THROTTLE_WAIT_MS = 200;
  // First render default wait after mounting (small wait for styled paint)
  const RENDER_WAIT_MS = 100;

  // Scrolling and other events that affect to viewport location have this safety margin
  // for lazy loading
  const NEAR_VIEWPORT_MARGIN = 50;

  class LazyLoadWithDimensionsComponent extends ReactComponent {
    constructor(props) {
      super(props);
      this.element = null;
      this.defaultRenderTimeout = null;
      this.afterRenderTimeout = null;

      this.state = { width: 0, height: 0 };

      this.handleWindowResize = throttle(this.handleWindowResize.bind(this), THROTTLE_WAIT_MS);
      this.isElementNearViewport = this.isElementNearViewport.bind(this);
      this.setDimensions = this.setDimensions.bind(this);
    }

    componentDidMount() {
      window.addEventListener('scroll', this.handleWindowResize);
      window.addEventListener('resize', this.handleWindowResize);
      window.addEventListener('orientationchange', this.handleWindowResize);

      this.defaultRenderTimeout = window.setTimeout(() => {
        if (this.isElementNearViewport(0)) {
          this.setDimensions();
        } else {
          const loadAfterInitialRendering = options.loadAfterInitialRendering;
          if (typeof loadAfterInitialRendering === 'number') {
            this.afterRenderTimeout = window.setTimeout(() => {
              window.requestAnimationFrame(() => {
                this.setDimensions();
              });
            }, loadAfterInitialRendering);
          }
        }
      }, RENDER_WAIT_MS);
    }

    componentWillUnmount() {
      window.removeEventListener('scroll', this.handleWindowResize);
      window.removeEventListener('resize', this.handleWindowResize);
      window.removeEventListener('orientationchange', this.handleWindowResize);
      window.clearTimeout(this.defaultRenderTimeout);

      if (this.afterRenderTimeout) {
        window.clearTimeout(this.afterRenderTimeout);
      }
    }

    handleWindowResize() {
      const shouldLoadToImproveScrolling = typeof options.loadAfterInitialRendering === 'number';
      if (this.isElementNearViewport(NEAR_VIEWPORT_MARGIN) || shouldLoadToImproveScrolling) {
        window.requestAnimationFrame(() => {
          this.setDimensions();
        });
      }
    }

    setDimensions() {
      this.setState(prevState => {
        const { clientWidth, clientHeight } = this.element || { clientWidth: 0, clientHeight: 0 };
        return { width: clientWidth, height: clientHeight };
      });
    }

    isElementNearViewport(safetyBoundary) {
      if (this.element) {
        const rect = this.element.getBoundingClientRect();
        const viewportHeight = window.innerHeight || document.documentElement.clientHeight;

        return (
          (rect.top >= 0 && rect.top <= viewportHeight + safetyBoundary) ||
          (rect.bottom >= -1 * safetyBoundary && rect.bottom <= viewportHeight)
        );
      }
      return false;
    }

    render() {
      const dimensions = this.state;
      const { width, height } = dimensions;
      const props = { ...this.props, dimensions };

      // lazyLoadWithDimensions HOC needs to take all given space
      // unless max dimensions are provided through options.
      const { maxWidth, maxHeight } = options;
      const maxWidthMaybe = maxWidth ? { maxWidth } : {};
      const maxHeightMaybe = maxHeight ? { maxHeight } : {};
      const style =
        maxWidth || maxHeight
          ? { width: '100%', height: '100%', ...maxWidthMaybe, ...maxHeightMaybe }
          : { position: 'absolute', top: 0, right: 0, bottom: 0, left: 0 };

      return (
        <div
          ref={element => {
            this.element = element;
          }}
          style={style}
        >
          {width !== 0 && height !== 0 ? <Component {...props} /> : null}
        </div>
      );
    }
  }

  LazyLoadWithDimensionsComponent.displayName = `lazyLoadWithDimensions(${Component.displayName ||
    Component.name})`;

  return LazyLoadWithDimensionsComponent;
};
