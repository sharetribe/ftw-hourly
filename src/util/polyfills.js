//
// Use common polyfill dependencies from package.json
//

// Smoothscroll
// TODO: scroll-behaviour smooth was added to Safari on March 2022.
//       We could consider removing this on 2023.
require('smoothscroll-polyfill').polyfill();

// To support browsers that do not have Intl.PluralRules (e.g IE11 & Safari 12-),
// - add npm packaged to package.json: "intl-pluralrules": "^1.3.1",
// - include this polyfill in your build:
//
// if (!Intl.PluralRules) {
//   require('intl-pluralrules');
// }

// To support  browsers that do not have Intl.RelativeTimeFormat (e.g IE11, Edge, Safari 12-),
// - add npm packaged to package.json: "@formatjs/intl-relativetimeformat": "^9.3.2",
// - include this polyfill in your build along with individual CLDR data for each locale you support:
//
// if (!Intl.RelativeTimeFormat) {
//   require('@formatjs/intl-relativetimeformat/polyfill');
//   require('@formatjs/intl-relativetimeformat/locale-data/en');

//   // By default, this library comes with en data. To load additional locale, you need include them on demand.
//   // e.g.
//   // require('@formatjs/intl-relativetimeformat/dist/locale-data/fr');
// }
