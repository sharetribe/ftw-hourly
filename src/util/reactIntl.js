import { func, shape } from 'prop-types';
import {
  IntlProvider,
  FormattedMessage,
  FormattedDate,
  createIntl,
  createIntlCache,
  injectIntl,
} from 'react-intl';

const intlShape = shape({
  formatDate: func.isRequired,
  formatMessage: func.isRequired,
  formatNumber: func.isRequired,
  formatPlural: func.isRequired,
  formatRelativeTime: func.isRequired,
  formatTime: func.isRequired,
});
export {
  IntlProvider,
  FormattedMessage,
  FormattedDate,
  createIntl,
  createIntlCache,
  injectIntl,
  intlShape,
};
