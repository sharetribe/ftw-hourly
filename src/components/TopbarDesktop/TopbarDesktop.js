import React, { useState, useEffect } from 'react';
import { bool, func, object, number, string } from 'prop-types';
import { FormattedMessage, intlShape } from '../../util/reactIntl';
import classNames from 'classnames';
import { ACCOUNT_SETTINGS_PAGES } from '../../routeConfiguration';
import { propTypes } from '../../util/types';
import {
  Avatar,
  InlineTextButton,
  Logo,
  Menu,
  MenuLabel,
  MenuContent,
  MenuItem,
  NamedLink,
  ListingLink,
  OwnListingLink,
} from '../../components';
import { TopbarSearchForm } from '../../forms';

import css from './TopbarDesktop.module.css';

const TopbarDesktop = props => {
  const {
    className,
    currentUser,
    currentPage,
    rootClassName,
    currentUserHasListings,
    currentUserListing,
    currentUserListingFetched,
    notificationCount,
    intl,
    isAuthenticated,
    onLogout,
    onSearchSubmit,
    initialSearchFormValues,
  } = props;
  const [mounted, setMounted] = useState(false);

  useEffect(() => {
    setMounted(true);
  }, []);

  const authenticatedOnClientSide = mounted && isAuthenticated;
  const isAuthenticatedOrJustHydrated = isAuthenticated || !mounted;

  const classes = classNames(rootClassName || css.root, className);

  const search = (
    <TopbarSearchForm
      className={css.searchLink}
      desktopInputRoot={css.topbarSearchWithLeftPadding}
      onSubmit={onSearchSubmit}
      initialValues={initialSearchFormValues}
    />
  );

  const notificationDot = notificationCount > 0 ? <div className={css.notificationDot} /> : null;

  const inboxLink = authenticatedOnClientSide ? (
    <NamedLink
      className={css.inboxLink}
      name="InboxPage"
      params={{ tab: currentUserHasListings ? 'sales' : 'orders' }}
    >
      <span className={css.inbox}>
        <FormattedMessage id="TopbarDesktop.inbox" />
        {notificationDot}
      </span>
    </NamedLink>
  ) : null;

  const currentPageClass = page => {
    const isAccountSettingsPage =
      page === 'AccountSettingsPage' && ACCOUNT_SETTINGS_PAGES.includes(currentPage);
    return currentPage === page || isAccountSettingsPage ? css.currentPage : null;
  };

  const profileMenu = authenticatedOnClientSide ? (
    <Menu>
      <MenuLabel className={css.profileMenuLabel} isOpenClassName={css.profileMenuIsOpen}>
        <Avatar className={css.avatar} user={currentUser} disableProfileLink />
      </MenuLabel>
      <MenuContent className={css.profileMenuContent}>
        <MenuItem key="EditListingPage">
          <OwnListingLink
            listing={currentUserListing}
            listingFetched={currentUserListingFetched}
            className={css.yourListingsLink}
          >
            <div>
              <span className={css.menuItemBorder} />
              {currentUserListing ? (
                <FormattedMessage id="TopbarDesktop.editYourListingLink" />
              ) : (
                <FormattedMessage id="TopbarDesktop.addYourListingLink" />
              )}
            </div>
          </OwnListingLink>
        </MenuItem>
        <MenuItem key="ProfileSettingsPage">
          <NamedLink
            className={classNames(css.profileSettingsLink, currentPageClass('ProfileSettingsPage'))}
            name="ProfileSettingsPage"
          >
            <span className={css.menuItemBorder} />
            <FormattedMessage id="TopbarDesktop.profileSettingsLink" />
          </NamedLink>
        </MenuItem>
        <MenuItem key="AccountSettingsPage">
          <NamedLink
            className={classNames(css.yourListingsLink, currentPageClass('AccountSettingsPage'))}
            name="AccountSettingsPage"
          >
            <span className={css.menuItemBorder} />
            <FormattedMessage id="TopbarDesktop.accountSettingsLink" />
          </NamedLink>
        </MenuItem>
        <MenuItem key="logout">
          <InlineTextButton rootClassName={css.logoutButton} onClick={onLogout}>
            <span className={css.menuItemBorder} />
            <FormattedMessage id="TopbarDesktop.logout" />
          </InlineTextButton>
        </MenuItem>
      </MenuContent>
    </Menu>
  ) : null;

  const signupLink = isAuthenticatedOrJustHydrated ? null : (
    <NamedLink name="SignupPage" className={css.signupLink}>
      <span className={css.signup}>
        <FormattedMessage id="TopbarDesktop.signup" />
      </span>
    </NamedLink>
  );

  const loginLink = isAuthenticatedOrJustHydrated ? null : (
    <NamedLink name="LoginPage" className={css.loginLink}>
      <span className={css.login}>
        <FormattedMessage id="TopbarDesktop.login" />
      </span>
    </NamedLink>
  );

  const listingLink =
    authenticatedOnClientSide && currentUserListingFetched && currentUserListing ? (
      <ListingLink
        className={css.createListingLink}
        listing={currentUserListing}
        children={
          <span className={css.createListing}>
            <FormattedMessage id="TopbarDesktop.viewListing" />
          </span>
        }
      />
    ) : null;

  const createListingLink =
    isAuthenticatedOrJustHydrated && !(currentUserListingFetched && !currentUserListing) ? null : (
      <NamedLink className={css.createListingLink} name="NewListingPage">
        <span className={css.createListing}>
          <FormattedMessage id="TopbarDesktop.createListing" />
        </span>
      </NamedLink>
    );

  return (
    <nav className={classes}>
      <NamedLink className={css.logoLink} name="LandingPage">
        <Logo
          format="desktop"
          className={css.logo}
          alt={intl.formatMessage({ id: 'TopbarDesktop.logo' })}
        />
      </NamedLink>
      {search}
      {listingLink}
      {createListingLink}
      {inboxLink}
      {profileMenu}
      {signupLink}
      {loginLink}
    </nav>
  );
};

TopbarDesktop.defaultProps = {
  rootClassName: null,
  className: null,
  currentUser: null,
  currentPage: null,
  notificationCount: 0,
  initialSearchFormValues: {},
  currentUserListing: null,
  currentUserListingFetched: false,
};

TopbarDesktop.propTypes = {
  rootClassName: string,
  className: string,
  currentUserHasListings: bool.isRequired,
  currentUserListing: propTypes.ownListing,
  currentUserListingFetched: bool,
  currentUser: propTypes.currentUser,
  currentPage: string,
  isAuthenticated: bool.isRequired,
  onLogout: func.isRequired,
  notificationCount: number,
  onSearchSubmit: func.isRequired,
  initialSearchFormValues: object,
  intl: intlShape.isRequired,
};

export default TopbarDesktop;
