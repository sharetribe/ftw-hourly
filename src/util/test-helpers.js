import React from 'react';
import mapValues from 'lodash/mapValues';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from '../store';
import { IntlProvider } from './reactIntl';

// In case you have translated the template and have new translations that
// are missing from the en translations file, the language for the tests can
// be changed here so that there are no missing translation keys in tests.
import messages from '../translations/en.json';

Enzyme.configure({ adapter: new Adapter() });

// Locale should not affect the tests. We ensure this by providing
// messages with the key as the value of each message.
const testMessages = mapValues(messages, (val, key) => key);

// Provide all the context for components that connect to the Redux
// store, i18n, router, etc.
export const TestProvider = props => {
  const store = configureStore();
  return (
    <IntlProvider locale="en" messages={testMessages} textComponent="span" timeZone="Etc/UTC">
      <BrowserRouter>
        <Provider store={store}>{props.children}</Provider>
      </BrowserRouter>
    </IntlProvider>
  );
};

// Use Enzyme's shallow rendering to render the given component to a
// JSON structure that can be used in snapshot tests. This doesn't
// render the children within the given component, only a
// representation of the child component and its props.
//
// Useful for snapshot testing components that contain shared
// components. With deep rendering, if the child component changes
// internally, the test for the given component would also fail. This
// avoids the problem by not rendering the full tree but only the
// relevant structure for the given component.
export const renderShallow = component => {
  return toJson(shallow(component));
};

// Fully render the given component to a JSON structure that can be
// used in snapshot tests.
export const renderDeep = component => {
  return toJson(mount(<TestProvider>{component}</TestProvider>), { mode: 'deep' });
};
