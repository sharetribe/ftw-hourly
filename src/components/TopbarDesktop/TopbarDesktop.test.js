import React from 'react';
import { fakeIntl } from '../../util/test-data';
import { renderDeep } from '../../util/test-helpers';
import TopbarDesktop from './TopbarDesktop';

const noop = () => null;

describe('TopbarDesktop', () => {
  it('data matches snapshot', () => {
    window.google = { maps: {} };
    const topbarProps = {
      isAuthenticated: true,
      currentUserHasListings: true,
      name: 'John Doe',
      onSearchSubmit: noop,
      intl: fakeIntl,
      onLogout: noop,
    };
    const tree = renderDeep(<TopbarDesktop {...topbarProps} />);
    delete window.google;
    expect(tree).toMatchSnapshot();
  });
});
