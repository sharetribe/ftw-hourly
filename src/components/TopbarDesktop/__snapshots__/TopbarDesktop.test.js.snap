// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopbarDesktop data matches snapshot 1`] = `
<nav
  className="root"
>
  <a
    className="logoLink NamedLink_active"
    href="/"
    onClick={[Function]}
    onMouseOver={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
    title={null}
  >
    <svg
      alt="TopbarDesktop.logo"
      className="logo"
      height="26"
      viewBox="0 0 141 26"
      width="141"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <g
          fillRule="nonzero"
        >
          <path
            d="M46.2 11.712l-2.184 5.664-2.472-6.48c-.288-.768-.36-1.032-.36-1.392 0-.648.408-.912 1.44-.936v-.576h-6.408v.552c1.128.096 1.344.36 2.16 2.28l4.2 10.032-.36.96c-.456 1.104-.792 1.416-1.536 1.416l-.141696-.003456c-.11904-.00624-.24768-.02208-.43992-.05652L39.672 23.088c-.48-.096-.744-.12-.96-.12-.624 0-.96.432-.96 1.248 0 1.104.6 1.656 1.752 1.656 1.488 0 2.352-.888 3.336-3.36l4.08-10.368c1.08-2.712 1.68-3.408 3.024-3.6v-.552h-5.232v.552c1.344.072 1.896.504 1.896 1.488 0 .456-.096.888-.408 1.68zm4.468 2.64c0 1.848.6 3.552 1.656 4.728 1.008 1.128 2.712 1.824 4.512 1.824 1.896 0 3.648-.72 4.752-1.944.984-1.104 1.584-2.88 1.584-4.752 0-3.984-2.448-6.576-6.216-6.576-1.824 0-3.456.624-4.488 1.704-1.128 1.2-1.8 3.048-1.8 5.016zm3.24-.12c0-1.9575.2109375-3.42984375.6723633-4.45658203L54.676 9.576c.408-.888 1.248-1.392 2.232-1.392 2.088 0 3.048 1.92 3.048 6.12 0 4.176-.936 6.048-3.024 6.048-2.064 0-3.024-1.968-3.024-6.12zm17.731657 4.344918L70.256 18.576l-.4964338-.0026951C67.56576 18.54752 67.064 18.328 67.064 17.544c0-.528.288-.864 1.104-1.296.768.192 1.44.264 2.136.264 1.416 0 2.712-.384 3.624-1.104.912-.72 1.44-1.92 1.44-3.264 0-1.32-.408-2.232-1.416-3.216h2.352V7.872l-2.568.096c-.6 0-.888-.024-1.656-.144l-.5241152-.08230453C70.9991111 7.66103704 70.6133333 7.632 70.184 7.632 67.04 7.632 65 9.456 65 12.264c0 1.128.36 2.136.984 2.736l.1463333.1393333C66.382 15.366 66.692 15.576 67.352 15.936l-1.224.768c-.984.624-1.224.984-1.224 1.704 0 .6.24 1.152.648 1.584.264.288.504.432 1.08.696l-.3962083.2268056C64.8896667 21.701 64.688 22.02 64.688 22.944c0 1.872 1.968 2.904 5.472 2.904 1.944 0 3.552-.36 4.704-1.056 1.056-.648 1.704-1.848 1.704-3.192 0-1.2-.528-2.088-1.44-2.52-.8664-.3648-1.6245-.49134-3.488343-.503082zM67.976 11.952c0-2.472.744-3.768 2.208-3.768 1.464 0 2.16 1.272 2.16 3.912 0 2.592-.696 3.864-2.16 3.864-1.44 0-2.208-1.416-2.208-4.008zm2.664 8.952c2.712 0 3.768.504 3.768 1.824 0 1.584-1.44 2.52-3.792 2.52-2.472 0-3.696-.96-3.696-2.856 0-.864.24-1.344.768-1.464l2.952-.024zm18.82-.456v-.576l-.3842164-.0113058c-1.0516003-.0535537-1.2210248-.332033-1.2443937-2.1367934l-.0018539-.9951328c.002304-.232128.008064-.557568.022464-.960768l.048-4.176c0-1.488-.288-2.352-.936-2.976-.72-.672-1.896-.984-3.552-.984-1.632 0-2.928.288-3.936.912-.936.576-1.488 1.392-1.488 2.232 0 .72.576 1.104 1.68 1.104.9487059 0 1.3659239-.2763737 1.4517435-1.02921067l.0490343-.56134489C81.2326667 9.66333333 81.316 9.34 81.516 9c.336-.576.84-.816 1.608-.816 1.392 0 1.84208.69522667 1.8705387 3.03956089L84.996 13.032c-1.488.072-2.76.24-3.936.456-2.568.504-3.84 1.776-3.84 3.84 0 2.16 1.416 3.528 3.696 3.528 1.56 0 2.856-.6 4.104-1.944l.036009.3724598C85.188 20.4384793 85.4498182 20.736 86.148 20.736l.2829653-.0062293C87.11376 20.7031467 88.2328 20.5936 89.46 20.448zm-4.464-6.864v1.848c0 1.368-.12 2.16-.432 2.784-.456.984-1.368 1.584-2.328 1.584-1.176 0-1.944-1.128-1.944-2.832 0-2.232 1.224-3.12 4.704-3.384z"
            fill="#000"
          />
          <path
            d="M94.888 8.76h2.904v-.768h-2.904l.0058776-1.55853061C94.9036735 5.36326531 94.9257143 4.78285714 94.96 4.08l-1.2.12c-.216 1.2-.456 1.896-.888 2.568-.552.864-1.224 1.224-2.28 1.272v.72h1.368l.001728 8.0108714c.012427 1.713484.0953155 2.2413025.358272 2.7891286.456.84 1.416 1.32 2.712 1.32 1.248 0 2.304-.384 3.288-1.2l-.336-.6c-.6.48-1.176.744-1.68.744s-.984-.36-1.176-.888c-.154-.44-.2273333-1.001-.2384861-1.9048333L94.888 8.76zm3.82-.768v.576l.3060683.0069331C100.4029844 8.62186889 100.508 8.89780645 100.508 10.896v6.648c0 2.136-.168 2.376-1.872 2.4v.576h6.672v-.576l-.3109938-.0077935c-1.3652263-.0531672-1.5493603-.3709749-1.5604358-2.1913593L103.436 12.336c0-1.536.048-3.216.144-4.656l-4.872.312zm1.56-4.104c0 .96.768 1.728 1.728 1.728s1.728-.768 1.728-1.728-.768-1.728-1.728-1.728-1.728.768-1.728 1.728zm10.588 3.792l-4.68.312v.576l.2985956.008448c1.26336.056448 1.464.388944 1.47672 2.099736L107.952 17.544c0 2.16-.168 2.376-1.824 2.4v.576h6.504v-.576l-.2892342-.0075131c-1.2703735-.0506408-1.4512367-.3501321-1.4621995-2.1894645L110.88 12.672c0-2.16 1.32-3.936 2.904-3.936 1.32 0 1.944 1.128 1.944 3.528v5.28c0 2.136-.168 2.376-1.752 2.4v.576h6.432v-.576l-.2935861-.0077935c-1.289344-.0531672-1.4708614-.3709749-1.4818467-2.1913593L118.632 12.48c0-.888.24-1.656.744-2.424.6-.912 1.272-1.32 2.136-1.32 1.416 0 1.992 1.008 1.992 3.552v5.256c0 2.136-.168 2.376-1.752 2.4v.576h6.504v-.576l-.30229-.0077935c-1.3272851-.0531672-1.5101108-.3709749-1.5211412-2.1913593L126.432 12.264c0-1.632-.192-2.52-.72-3.24-.648-.912-1.728-1.392-3.12-1.392-1.8 0-3.072.768-4.128 2.52-.432-1.704-1.56-2.52-3.408-2.52-1.8 0-3.216.84-4.176 2.424l-.024-2.376zm29.884 6.072c-.048-1.512-.168-2.232-.432-2.952-.792-2.064-2.52-3.168-4.968-3.168-3.648 0-6.024 2.64-6.024 6.672 0 4.008 2.448 6.576 6.264 6.576 2.472 0 4.08-1.008 5.28-3.288l-.6-.384c-.888 1.752-2.304 2.736-3.96 2.736-1.464 0-2.568-.792-3.264-2.376-.432-1.008-.6-2.04-.624-3.816h8.328zm-8.304-.696c.096-1.248.24-2.016.504-2.76.504-1.392 1.296-2.112 2.328-2.112.936 0 1.608.6 2.064 1.872.264.744.36 1.488.408 3h-5.304z"
            fill="#0095B3"
          />
        </g>
        <g
          stroke="#0095B3"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1.5"
        >
          <path
            d="M14.0221667 14.9028333c-2.123449-1.7178872-3.3997596-4.2705084-3.5-6.99999997C10.5891066 5.44596767 11.6329727 3.11721418 13.4225 1.4325c.3334333-.32683002.8670667-.32683002 1.2005 0 1.7886677 1.68521534 2.8320183 4.01375565 2.8991667 6.47033333-.1002404 2.72949157-1.376551 5.28211277-3.5 6.99999997z"
          />
          <path
            d="M17.43 6.67783333C19.1014265 5.54101183 21.0786397 4.93849069 23.1 4.95c.4796887.00063444.8689352.3883177.8715.868.0697835 2.45026057-.8386379 4.82752063-2.5246667 6.6068333-2.0009973 1.8590387-4.7084726 2.7613887-7.4246666 2.4745M10.6166667 6.69066667C8.9359545 5.53533534 6.93938056 4.92739894 4.9 4.95c-.48033591-.0000107-.87062682.38767479-.87383333.868-.07166144 2.4509653.835542 4.82959311 2.52116666 6.6103333 2.01953017 1.8618845 4.74324717 2.7635551 7.47483337 2.4745M22.75 16.0625c0 2.4161667-3.9176667 4.375-8.75 4.375-4.83233333 0-8.75-1.9541667-8.75-4.375"
          />
          <path
            d="M26.0283333 12.559c.701412 1.0334801 1.0821012 2.2510363 1.0943334 3.5 0 4.8323333-5.8765 8.75-13.125 8.75-7.24850003 0-13.12500003-3.9141667-13.12500003-8.7465.01168117-1.248701.39154021-2.4661979 1.092-3.5"
          />
        </g>
      </g>
    </svg>
  </a>
  <form
    action="/"
    className="searchLink"
    method="post"
    onMouseOver={[Function]}
    onSubmit={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="root topbarSearchWithLeftPadding"
    >
      <div
        className="desktopIcon"
      >
        <svg
          className="iconSvg"
          height="22"
          viewBox="0 0 21 22"
          width="21"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            className="iconSvgGroup"
            fill="none"
            fillRule="evenodd"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            transform="matrix(-1 0 0 1 20 1)"
          >
            <path
              d="M13 14l5.241 5.241"
            />
            <circle
              cx="7.5"
              cy="7.5"
              r="7.5"
            />
          </g>
        </svg>
      </div>
      <input
        autoComplete="off"
        autoFocus={false}
        className="desktopInput"
        disabled={false}
        name="location"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="TopbarSearchForm.placeholder"
        type="search"
        value=""
      />
    </div>
  </form>
  <a
    className="inboxLink"
    href="/inbox/sales"
    onClick={[Function]}
    onMouseOver={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
    title={null}
  >
    <span
      className="inbox"
    >
      <span>
        TopbarDesktop.inbox
      </span>
    </span>
  </a>
  <div
    className="root"
    onBlur={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <button
      className="root profileMenuLabel"
      onBlur={[Function]}
      onClick={[Function]}
    >
      <div
        className="root avatar"
        title=""
      >
        <span
          className="initials"
        />
      </div>
    </button>
    <div
      className="root profileMenuContent isClosed"
      style={
        Object {
          "left": "-10000px",
        }
      }
    >
      <ul
        className="content"
      >
        <li
          className="root"
          role="menuitem"
        />
        <li
          className="root"
          role="menuitem"
        >
          <a
            className="profileSettingsLink"
            href="/profile-settings"
            onClick={[Function]}
            onMouseOver={[Function]}
            onTouchStart={[Function]}
            style={Object {}}
            title={null}
          >
            <span
              className="menuItemBorder"
            />
            <span>
              TopbarDesktop.profileSettingsLink
            </span>
          </a>
        </li>
        <li
          className="root"
          role="menuitem"
        >
          <a
            className="yourListingsLink"
            href="/account"
            onClick={[Function]}
            onMouseOver={[Function]}
            onTouchStart={[Function]}
            style={Object {}}
            title={null}
          >
            <span
              className="menuItemBorder"
            />
            <span>
              TopbarDesktop.accountSettingsLink
            </span>
          </a>
        </li>
        <li
          className="root"
          role="menuitem"
        >
          <button
            className="logoutButton inlineTextButton"
            disabled={false}
            onClick={[Function]}
          >
            <span
              className="menuItemBorder"
            />
            <span>
              TopbarDesktop.logout
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
